### DISCUSSION TITLE
Support updating versions on npx calls within the package.json

### DISCUSSION THREAD

**Discussion by jase88** [renovatebot/renovate#14390] [Suggest an Idea]
### What would you like Renovate to be able to do?

There might be cases where an npm dependency should not be part of the dependencies within the `package.json`.
This could be mainly CLIs, but there may be other use-cases as well.

Renovate should be able to scan npx usages within the scripts section in the `package.json` and update the used version of a package.

Example
```
{
  "name": "some-project",
  "scripts": {
    "cap": "npx --yes --package @capacitor/cli@3.4.0 cap"
  },
  "dependencies": {
    "something": "1.2.3"
  }
}
```
In this case renovate should be able to update the package `@capacitor/cli`.

### If you have any ideas on how this should be implemented, please tell us here.

- scan the scripts section property of the `package.json` and identify used package versions
- update the version string on the scripts section if necessary

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by viceice** [comment]
Too complicated, use regex manager.

**Reply by rarkins** [reply]
I agree, I think it would be best to implement as a regex manager. If you use it commonly then you can make it part of your central config or as its own preset

**Reply by jase88** [reply]
thanks didn't know about regex manager functionality.
I close this feature request ticket

**Reply by soullivaneuh** [reply]
I understand completely that kind of implementation could be very complex because it requires to be able to match that kind of CLI calls over many file format. Here is another example with a `.gitlab-ci.yml` configuration under the `script` section:

```yaml
lint:npm:depcheck:
  script:
    # ...
    - npx depcheck@1.4.3
```

Ref: https://gitlab.com/ss-open/ci/recipes/-/blob/1cb96978a33a3ee21150dbbb89e3b4708ff2b311/stages/lint/npm:depcheck/.gitlab-ci.yml#L1-10

Plus, we are talking about `npx` call here, but we will be facing to alternative such as `bunx`.

However, the needs are common across project and communities, meaning I am not the only one having the need to have an automated update for `npx` calls trough GitLab CI configuration.

May it be a possibility to setup a space on the official repository where we can propose such contributions? If not, shall you elaborate a bit the reason? :thinking: 

Best regards

**Reply by rarkins** [reply]
You can solve with the Regex manager and it does not need new code

**Reply by soullivaneuh** [reply]
@rarkins Yes, I understood that. The main subject of my comment is:

> May it be a possibility to setup a space on the official repository where we can propose such contributions? If not, shall you elaborate a bit the reason? 🤔

Meaning, a space where we can propose Regex models to be use by the community. Maybe not on the renovate repository but a third party one dedicated for that?

**Reply by rarkins** [reply]
Yes, here: https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/custom-managers.ts

**Reply by soullivaneuh** [reply]
@rarkins Thanks! Are contribution like for this subject accepted or is it just for your internal usage?

**Reply by rarkins** [reply]
We will accept presets as long as we think they are applicable to a reasonably large number of users (and not just personal presets or company internal). "internal" means they are built into Renovate by default, and not that they're internal to us as Renovate maintainers
